ccmlib/scylla_repository.py

```diff
@@ -234,15 +235,40 @@ def setup(version, verbose=True, skip_downloads=False):
               - release:4.2.1
               - release:2020.1
               - release:2020.1.5
+            flags can be added to the version using [(flag(=value)?)+]
+              - supported flags are:
+                - product=(scylla|scylla-enterprise)
```

Comment:
Don't we already have enough information to determine scylla vs scylla-enterprise without having to specify it?

Reply:
it's obvious for `unstable/enterprise`, but for `unstable/branch-2023.1` we'll have to second-guess the branch naming conventions. That's possible of course and we already do it elsewhere, not that I like it very much...

Reply:
Some cases, like for `byo` I think we wouldn't be able to infer the product from the scylla_version.

Reply:
Speaking of that, how can we represent a version for e.g. `https://downloads.scylladb.com/testing/unstable/scylla-enterprise/enterprise-2024.1/relocatable/byo/2024-07-23T20%3A47%3A40Z/`
Maybe we should support a url (`s3://` or `https://`) in addition to the current shorthand forms.

Reply:
Something like `testing/enterprise-2024.01:byo/2024-07-23T20:47:40Z`
Even though what is before the `/` is ignored:
https://github.com/scylladb/scylla-ccm/blob/9ce5d753d213e7fc682bf0d121b8c59b597199a0/ccmlib/scylla_repository.py#L274

We would just need to add another URL to this list:
https://github.com/scylladb/scylla-ccm/blob/9ce5d753d213e7fc682bf0d121b8c59b597199a0/ccmlib/scylla_repository.py#L32-L35

Something like `https://s3.amazonaws.com/downloads.scylladb.com/testing/unstable/scylla-enterprise/{0}/relocatable/{1}/}`

Of course, there are other cases in the testing subfolder that would need to be covered as well.

Reply:
We do support download from a full path of the unified package, it's just not via scylla_version parameter.

That would be enough, we don't need so many complications to this one.

As for the product, we do estimate it good enough base on release or branch name. I haven't used it for quite some time now, I think it can be removed, and not introduced as a new option.

Reply:
Let's postpone further cleanups till after we merge this PR
I think it has enough value on its own already.